tests/serverpod_test_server/test/database_operations/generated_operations/batch_interactions_test.dart

```diff
@@ -0,0 +1,640 @@
+import 'dart:typed_data';
+
+import 'package:serverpod_test_client/serverpod_test_client.dart';
+import 'package:test/test.dart';
+
+import '../../config.dart';
+
+void main() {
+  var client = Client(serverUrl);
+
+  tearDown(() async => await client.databaseBatch.deleteAll());
+  group('Given an empty database', () {
+    tearDown(() async => await client.databaseBatch.deleteAll());
+    test(
+        'when batch inserting then all the entries are created in the database.',
+        () async {
+      var data = <UniqueData>[
+        UniqueData(number: 1, email: 'info@serverpod.dev'),
+        UniqueData(number: 2, email: 'dev@serverpod.dev'),
+      ];
+
+      var inserted = await client.databaseBatch.batchInsert(data);
+
+      expect(inserted, hasLength(2));
+
+      var simpleList = await client.databaseBatch.findAll();
+
+      expect(inserted.first.id, equals(simpleList.first.id));
+      expect(inserted.last.id, equals(simpleList.last.id));
+    });
+
+    test(
+        'when batch inserting with one failing row then no entries are created in the database.',
+        () async {
+      var data = <UniqueData>[
+        UniqueData(number: 2, email: 'info@serverpod.dev'),
+        UniqueData(number: 2, email: 'dev@serverpod.dev'),
+        UniqueData(number: 2, email: 'dev@serverpod.dev'),
+      ];
+
+      expect(
+        client.databaseBatch.batchInsert(data),
+        throwsA(isA<ServerpodClientException>()),
+      );
+
+      var first = await client.databaseBatch.findByEmail('info@serverpod.dev');
+      expect(first, isNull);
+
+      var second = await client.databaseBatch.findByEmail('dev@serverpod.dev');
+      expect(second, isNull);
+    });
+
+    test('when batch inserting with an id defined then the id is ignored.',
+        () async {
+      const int max = 1 >>> 1;
```

Comment:
Question: For my knowledge, you are trying to define int max here right? Doesn't dart have an int max const value?

Reply:
yes, and dart does not have an int max const but weirdly enough there is a Double.max value.